build/parse.y

```diff
@@ -1062,25 +1062,36 @@ func forceMultiLineComprehension(start Position, expr Expr, clauses []Expr, end
 	return previousEnd.Line != end.Line
 }
 
-// extractTrailingComment extracts a trailing comment from a block statement
-// and returns the comment (or nil)
-func extractTrailingComment(stmt Expr) *CommentBlock {
+// extractTrailingComments extracts trailing comments from a block statement
+// and returns the comments. The comments can be either CommentBlock statements
+// or After-comments for a statement of a different type.
+func extractTrailingComments(stmt Expr) []Expr {
 	body := getLastBody(stmt)
+	var comments []Expr
 	if body != nil && len(*body) > 0 {
-		lastStmt := (*body)[len(*body)-1]
-		if cb, ok := lastStmt.(*CommentBlock); ok {
-			// Remove the comment block
-			*body = (*body)[:len(*body)-1]
-			return cb
+		// Detach and return all trailing comment blocks
+		for i := len(*body)-1; i >= 0; i-- {
+			if cb, ok := (*body)[i].(*CommentBlock); ok {
```

Comment:
reverse the condition, remove the continue

Reply:
If I reverse the condition I'll need to assign `cb` to a variable that's defined outside of this block. Will it be more readable?

Reply:
I think so.

Reply:
Done.